trilitech/jstz
# Cli & Api & Proto: Introduce `ConsoleLog` trait

# Context
Currently the console logging methods aren't flexible. We also had an issue where the loggings method gets reverted (e.g. calling a contract in repl would change the logging method from `pretty` to `json`. 

**Related Tasks**: [Task](https://app.asana.com/0/1205770721173533/1206083271582090) 

# Description
This PR
* introduces the `JsLog` trait that provides a way to control how logging is done in the js console object. There are two implementations: `JsonLogger` for proto and `PrettyLog` for repl.
* `set_js_logger` can be called to set the type of logger before running the script 
* Introduced a new global variable `TraceData` in HostDefined that contains both the`contract_address` and `operation_hash`. This way we don't have to store those fields in the ConsoleApi or ContractApi
* Pretty print the log in logs trace <sf>

# Manually testing the PR
Proto: 
* start the sandbox and jstz_node
* deploy a contract
* run `jstz logs trace <sf>` to listen to the logs
* run the deployed contract to see if the logs are properly working

https://github.com/trilitech/jstz/assets/128799322/0fbd007f-50e4-4f02-bbde-99514956c265


Repl: 
* console.log / error / info  in repl
